configs/project.go

```diff
@@ -1,55 +1,189 @@
 package configs
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
 	"github.com/railwayapp/cli/entity"
 	"github.com/railwayapp/cli/errors"
 )
 
-func (c *Configs) GetProjectConfigs() (*entity.ProjectConfig, error) {
-	var cfg entity.ProjectConfig
+// MigrateLocalProjectConfig moves a local project config
+// to the global config and removes the local .railway directory
+// if it exists
+func (c *Configs) MigrateLocalProjectConfig() error {
+	// Get local config directory
+	projectDir, err := filepath.Abs(filepath.Dir(filepath.Dir(c.projectConfigs.configPath)))
+	if err != nil {
+		return err
+	}
+
+	if _, err := os.Stat(projectDir); os.IsNotExist(err) {
+		// Local project directory does not exist
+		return nil
+	}
 
+	// Read local project config
+	var cfg entity.ProjectConfig
 	if err := c.unmarshalConfig(c.projectConfigs, &cfg); err != nil {
+		return err
+	}
+
+	// Save project config to root config
+	cfg.ProjectPath = strings.ToLower(projectDir)
+	if err = c.SetProjectConfigs(&cfg); err != nil {
+		return err
+	}
+
+	// Delete local config directory
+	if err = os.RemoveAll(fmt.Sprintf("%s/.railway", projectDir)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Configs) getCWD() (string, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+	cwd = strings.ToLower(cwd)
+
+	return cwd, nil
+}
+
+func (c *Configs) GetProjectConfigs() (*entity.ProjectConfig, error) {
+	c.MigrateLocalProjectConfig()
+
+	userCfg, err := c.GetRootConfigs()
+	if err != nil {
+		return nil, errors.ProjectConfigNotFound
+	}
+
+	// lookup project in global config based on pwd
+	cwd, err := c.getCWD()
+	if err != nil {
+		return nil, err
+	}
+
+	var longestPath = -1
+	var pathMatch = ""
+	for path := range userCfg.Projects {
+		// use exact match if it exsits
+		if path == cwd {
+			longestPath = len(path)
+			pathMatch = path
+			break
+		}
+
+		// check if parent path matches
+		var matches = strings.HasPrefix(cwd, fmt.Sprintf("%s/", path))
+		if matches && len(path) > longestPath {
+			longestPath = len(path)
+			pathMatch = path
+		}
```

Comment:
These could probably be combined no? Simply use string contains, or maybe even prefix will handle it

Reply:
The reason I separated it is because of the trailing slash. The current directory does not have a trailing slash but we want to make sure that `/this/path` does not match `/this/pathalso`.

Thinking of it now I can just add a trailing slash to the current directory path and combine the two checks :)